packages/react-menu/src/utils/useOnMenuEnter.ts

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import { useEventCallback, UseOnClickOrScrollOutsideOptions } from '@fluentui/react-utilities';
+import { elementContains } from '@fluentui/react-portal';
+
+/**
+ * Name of the custom event
+ */
+export const MENU_ENTER_EVENT = 'fuimenuenter';
```

Comment:
Would be nice to have an RFC for making this a pattern for custom events that can be public.